pennylane/devices/qubit_mixed/apply_operation.py

```diff
@@ -163,7 +165,24 @@ def _get_num_wires(state, is_state_batched):
     """
     For density matrix, we need to infer the number of wires from the state.
     """
-    return (math.ndim(state) - is_state_batched) // 2
+
+    s = qml.math.shape(state)
+    flat_size = 1
+    for dim in s:
+        flat_size *= dim
+
+    if is_state_batched:
+        batch_size = s[0]
+    else:
+        batch_size = 1
+
+    total_dim = flat_size // batch_size
```

Comment:
```suggestion
    shape = qml.math.shape(state)
    batch_size = shape[0] if is_state_batched else 1
    total_dim = math.prod(shape) // batch_size
```

Isn't this the same?